internal/controller/purge_controller.go

```diff
@@ -163,6 +158,17 @@ func (r *PurgeReconciler) dropPurgeFinalizer(ctx context.Context, kyma *v1beta2.
 	return nil
 }
 
+func (r *PurgeReconciler) deadlineReached(kyma *v1beta2.Kyma) (bool, time.Duration) {
```

Comment:
I moved it to a seperate function since we hit the funlen of the Reconcile method, plus to simplify complexity. I would leave it like this since this is an internal function of this pkg and not a function which should be reused outside of it.